Remove some .data and clearly mark why we keep some

albanD opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31482 remove simple .data from torch/nn
* #31481 remove .data from torch/nn doc
* #31480 remove .data from torch/jit
* **#31479 Remove some .data and clearly mark why we keep some**

Differential Revision: [D19303185](https://our.internmc.facebook.com/intern/diff/D19303185)

albanD committed:
Remove some .data and clearly mark why we keep some

[ghstack-poisoned]

gchanan commented on torch/testing/__init__.py:
```diff
@@ -83,7 +82,9 @@ def make_non_contiguous(tensor):
-    return input
+
+    # Use .data here to hide the view relation between input and other temporary Tensors
+    return input.data
```
why can't this one use no_grad?

albanD replied:
Unfortunately no because we track views even when grad mode is disabled at the moment.
This will be fixed at the same time as the bug in Function that return view of intermediary results.

gchanan replied:
but what makes you realize you need to add it here?  Like, which call is a problem?  Is it coming from gradcheck?

albanD replied:
This is necessary for `gradgradcheck` when we remove the `.data` in the numerical gradient for functions for which the version bumping is not an issue.
The idea is that:
- This will be required when we fix the `.data` in gradcheck.
- It took me half an hour to find that this was the issue. And this looks like a good way to document this finding.

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 179650f:


* **1/1** broken upstream at merge base fc3103b (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=fc3103b1162a3312e4ad74005d0b698a333c0750&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
  <details>
  <summary>
  You may want to rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  
* **0/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=179650fc5e7204e5fc58dddb6e639cb0f0cfce1e).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_linux_xenial_py3_6_gcc5_4_test](https://dr.pytorch.org/build-details.html?build_id=77616413)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

albanD committed:
Update on "Remove some .data and clearly mark why we keep some"




[ghstack-poisoned]

albanD committed:
Update on "Remove some .data and clearly mark why we keep some"




[ghstack-poisoned]

albanD committed:
Update on "Remove some .data and clearly mark why we keep some"


Differential Revision: [D19303185](https://our.internmc.facebook.com/intern/diff/D19303185)

[ghstack-poisoned]

albanD commented:
Bot mixup, this pull request was merged in 26621d1.
